### PR TITLE
Generate Lambda docs, add index page with example

### PR DESCRIPTION
**Pull request by evanphx**
Get those lambda docs out there!

### CODE REVIEW

File: website/content/partials/components/platform-aws-lambda.mdx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+- Type: **int**
+- **Optional**
+- Default: 60
+
+### Output Attributes
+
+Output attributes can be used in your `waypoint.hcl` as [variables](/docs/waypoint-hcl/variables) via [`artifact`](/docs/waypoint-hcl/variables/artifact) or [`deploy`](/docs/waypoint-hcl/variables/deploy).
+
+#### func_arn
```
Comment (briancain) [inline website/content/partials/components/platform-aws-lambda.mdx:63]: Should the rest of these options have some sort of description/example here?

Comment (evanphx) [reply website/content/partials/components/platform-aws-lambda.mdx:63]: So those are not available in the documentation framework to be described, as such nothing has info on them atm.

File: website/content/partials/components/platform-aws-lambda.mdx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+An IAM Role specified by ARN that will be used by the Lambda at execution time.
+
+- Type: **string**
+- **Optional**
+- Default: created automatically
+
+#### memory
+
+The amonut of memory, in megabytes, to assign the function.
```
Comment (krantzinator) [inline website/content/partials/components/platform-aws-lambda.mdx:45]: I know these are generated so you'll have to edit the source and not directly commit this, but FYI
```suggestion
The amount of memory, in megabytes, to assign the function.
```

File: website/content/plugins/aws-lambda.mdx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+## Image Support Only
+
+The AWS Lambda deploy plugin only supports AWS Lambda deployments that utilize ECR images
+rather than .zip files.
+
+We recommend that the `docker` build plugin is used to generate these images.
+
+Below is a simple Dockerfile, handler code, and waypoint.hcl for build, deploying, and
+releasing using Waypoint.
```
Comment (krantzinator) [inline website/content/plugins/aws-lambda.mdx:18]: Make these the same tense "build, deploy, and release" or "building, deploying, and releasing"

**Review by krantzinator — CHANGES_REQUESTED** [review]

**Review by krantzinator — APPROVED** [review]

**Review by briancain — APPROVED** [review]

**Pull request merged**